### DISCUSSION TITLE
is there a way to refresh rules()

### DISCUSSION THREAD

**Discussion by sathia-musso** [livewire/livewire#3408] [1. Help] [answered]
Hi, I'm appending form elements as the user selects between different options. 
Since I don't know which options the user will select beforehand I want to be able to inject rules to the rules() function before rendering the new components on the page.
I haven't found a way to successfully do it, is there a way? 

thanks!

**Comment by devcircus** [comment]
Couldn't you conditionally return the necessary rules from the rules function based on the requirements derived from the current state of the component?

**Comment by sathia-musso** [comment]
> Couldn't you conditionally return the necessary rules from the rules function based on the requirements derived from the current state of the component? 

I believe rules() is invoked only once upon mount.

**Comment by joshhanley** [comment, marked as answer]
@sathio it's not actually called upon mount at all, but the rules method is called each time it is used, so you should be able to put any necessary conditional logic into the rules method like @devcircus suggested.

The catch is though, you don't want to do any manual DB queries inside your rules method otherwise they will be called multiple times per request, as the rules are used for both validation and dehydration of eloquent models.

Hope this helps!

**Reply by sathia-musso** [reply]
Hi, thank you, I was in fact, wrong, I've looked through the code again and found out how it works, I managed to have it working in the end. thanks again!

**Reply by joshhanley** [reply]
No worries 🙂

**Comment by ChrisSantiago82** [comment]
Not sure if that helps. You could declare the rules variable public and you could change the rules array when the option is selected
